### ISSUE TITLE
Ragged paged-MQA logits use incorrect causal bounds for multi-token queries

### ISSUE THREAD

**Issue by morluto** [ROCm/aiter#4290]
## Problem

The final-logits and stage-one ragged paged-MQA kernels use:

```python
context_length - pid_next_n
```

as their causal upper bound. For query `i` among `next_n` appended tokens, the absolute query position is:

```text
context_length - next_n + i
```

The current predicate is therefore incorrect whenever `next_n > 1`.

## Reproduction

For `context_length=8` and `next_n=4`, the four queries occupy positions `4, 5, 6, 7`.

| Query | Expected maximum KV | Current maximum within context | Result |
|---:|---:|---:|---|
| 0 | 4 | 7 | exposes positions 5-7 |
| 1 | 5 | 7 | exposes positions 6-7 |
| 2 | 6 | 6 | correct |
| 3 | 7 | 5 | masks positions 6-7 |

Query 0 can also emit a finite zero at `k == context_length`, because the masked KV load zero-fills data but does not mask the resulting logit.

## Proof

Let `L=context_length`, `n=next_n`, and `i=pid_next_n`.

```text
query_position(i) = L - n + i
valid KV positions = { k | 0 <= k <= L - n + i }
```

The current kernel instead uses `k <= L-i`.

- For `i=0`, the old maximum within the real context is `L-1`, which is `n-1` positions beyond the correct maximum.
- For `i>0`, the old maximum is `L-i`; subtracting the correct maximum gives `n-2i`. Positive values expose future positions, while negative values mask valid positions.

## Expected behavior

The causal predicate should be:

```python
k <= context_length - next_n + pid_next_n
```

## Scope

Both kernels are publicly exposed. The benchmark retains a ragged reference while currently selecting only the non-ragged implementation.

## Success criteria

Regression tests through both public wrappers should verify last finite positions `4, 5, 6, 7` and require every later position to be `-inf`.
